src/inmanta/agent/agent.py

```diff
@@ -372,40 +372,58 @@ class DeployRequestAction(str, enum.Enum):
     """ interrupt the current deploy: cancel it, start the new one and restart the current deploy when the new one is done"""
 
 
+# Two letter abbreviations to make table line up
+# N normal
+# P periodic
+# F full
+# I incremental
+NF = (True, False)
+NI = (False, False)
+PF = (True, True)
+PI = (False, True)
+# shorten table
+ignore = DeployRequestAction.ignore
+terminate = DeployRequestAction.terminate
+defer = DeployRequestAction.defer
+interrupt = DeployRequestAction.interrupt
+
 # This matrix describes what do when a new DeployRequest enters before the old one is done
-# Format is old_is_repair, old_is_periodic, new_is_repair, new_is_periodic
+# Format is (old_is_repair, old_is_periodic), (new_is_repair, new_is_periodic)
 # The underlying idea is that
 # 1. periodic deploys have no time pressure, they can be delayed
 # 2. non-periodic deploy should run as soon as possible
 # 3. non-periodic incremental deploy take precedence over repairs (as they are smaller)
 # 4. periodic deploys should not interrupt each other to prevent restart loops
+# 5. Periodic repairs take precedence over periodic incremental deploys.
```

Comment:
I think the fallback rule is still missing: if none of these conditions match, meaning `(NF, NF)` or `(NI, NI)`, we pick the newest. Or do you see this as a consequence of rule 2?